test/libyul/Parser.cpp

```diff
@@ -573,9 +568,9 @@ BOOST_AUTO_TEST_CASE(customSourceLocations_reference_original_sloc)
 	BOOST_REQUIRE(holds_alternative<VariableDeclaration>(result->statements.at(0)));
 	VariableDeclaration const& varDecl = get<VariableDeclaration>(result->statements.at(0));
 
-	// -1 points to original source code, which in this case is `sourceText` (which is also
+	// -1 points to original source code, which in this case is `"source0"` (which is also
 	// available via `0`, that's why the first @src is set to `1` instead.)
-	CHECK_LOCATION(varDecl.debugData->location, sourceText, 10, 20);
+	CHECK_LOCATION(varDecl.debugData->location, "", 10, 20);
```

Comment:
@christianparpart I'm not really sure why this is not "source0" here... Can you answer this question?

Reply:
I think it is empty because it's the source location of the yul source (the string in the unit test), not the one mentioned above.

It is at least going into the right code path:
![image](https://user-images.githubusercontent.com/56763/125442571-35a58b7a-fd9b-4294-adb1-7e3e59b99107.png)

So the question is why is `ParserBase::currentLocation().sourceName` an empty string.

Reply:
Fix is in https://github.com/ethereum/solidity/pull/11596/files#r668745589